pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -277,3 +277,16 @@ func activeThreshold(pa *pav1alpha1.PodAutoscaler) int {
 
 	return int(min)
 }
+
+// resolveScrapeTarget returns metric service name to be scraped based on TBC configuration
+// TBC == -1 => activator in path, don't scrape the service
+func resolveScrapeTarget(ctx context.Context, pa *pav1alpha1.PodAutoscaler, metricSN string) string {
```

Comment:
```suggestion
func resolveScrapeTarget(ctx context.Context, pa *pav1alpha1.PodAutoscaler) string {
```

You can just get the MetricsServiceName from the PA's status directly below.

Reply:
Fixed in 7b1feced1c20f12f662502628344d8087c3978f6.